src/app/modules/myinfo/myinfo.util.ts

```diff
@@ -306,16 +312,36 @@ const isMyInfoFormWithEsrvcId = <F extends IFormSchema>(
 }
 
 /**
- * Type guard for MyInfo cookie.
+ * Type guard for MyInfo login cookie.
  * @param cookie Unknown object
  */
-export const isMyInfoCookie = (
+export const isMyInfoLoginCookie = (
   cookie: unknown,
-): cookie is MyInfoCookiePayload => {
-  if (cookie && typeof cookie === 'object' && hasProp(cookie, 'state')) {
+): cookie is MyInfoLoginCookiePayload => {
+  if (
+    cookie &&
+    typeof cookie === 'object' &&
+    hasProp(cookie, 'uinFin') &&
+    typeof cookie.uinFin === 'string'
+  ) {
+    return true
+  }
+  return false
+}
```

Comment:
nit: simplify to returning conditional immediately

```suggestion
return (
  !!cookie &&
  typeof cookie === 'object' &&
  hasProp(cookie, 'uinFin') &&
  typeof cookie.uinFin === 'string'
  ) 
}
```

Reply:
676072bc